proposals/0292-package-registry-service.md

```diff
@@ -38,77 +43,223 @@ it's not well-suited to this kind of workflow for the following reasons:
+to fetch a list of available releases
+from the configured registry.
+If a release is found that satisfies the declared version requirement
+(for example, `.upToNextMinor(from: "1.1.0")`),
+Swift Package Manager sends a
+`GET` request to `/{scope}/{name}/{version}/Package.swift`
+to fetch the manifest for that release.
+This process continues with the package manifests of each dependency,
+each of their respective dependencies,
+and so on.
+Once the dependency graph is [resolved](#dependency-graph-resolution),
+Swift Package Manager downloads the
+[source archive](#archive-source-subcommand) for each dependency
+by sending a `GET` request to `/{scope}/{name}/{version}.zip`.
 
 ## Detailed design
 
 ### Package registry service
 
-A package registry service implements REST API endpoints
+A package registry service implements the following REST API endpoints
 for listing releases for a package,
 fetching information about a release,
-downloading the source archive for a release,
-and publishing a new release of a package.
+and downloading the source archive for a release:
 
-| Method | Path                                                 | Description                                      |
-| ------ | ---------------------------------------------------- | ------------------------------------------------ |
-| `GET`  | `/{package}`                                         | List package releases                            |
-| `GET`  | `/{package}/{version}`                               | Fetch metadata for a package release             |
-| `GET`  | `/{package}/{version}/Package.swift{?swift-version}` | Fetch manifest for a package release             |
-| `GET`  | `/{package}/{version}.zip`                           | Download source archive for a package release    |
-| `PUT`  | `/{package}/{version}{?commit,branch,tag,path,url}`  | Publish a package release or update its metadata |
+| Method | Path                                                      | Description                                     |
+| ------ | --------------------------------------------------------- | ----------------------------------------------- |
+| `GET`  | `/{scope}/{name}`                                         | List package releases                           |
+| `GET`  | `/{scope}/{name}/{version}`                               | Fetch metadata for a package release            |
+| `GET`  | `/{scope}/{name}/{version}/Package.swift{?swift-version}` | Fetch manifest for a package release            |
+| `GET`  | `/{scope}/{name}/{version}.zip`                           | Download source archive for a package release   |
+| `GET`  | `/identifiers{?url}`                                      | Lookup package identifiers registered for a URL |
 
 A formal specification for the package registry interface
 is provided alongside this proposal.
 In addition,
-an OpenAPI (v3) document and a reference implementation
+an OpenAPI (v3) document
```

Comment:
Should this link to the OpenAPI document? I seem to have lost the link and I wanted to review it.

Reply:
The server specification and OpenAPI document will be located in the [`@apple/swift-package-manager` repository](https://github.com/apple/swift-package-manager/pull/3364). ([Direct link to the spec](https://github.com/apple/swift-package-manager/blob/e593c129d6438c509b4220321cbd26b804e7733e/Documentation/Registry.md)).

Sorry for the inconvenience here — I couldn't think of a good way to durably link between these documents while they were still in-flight.